File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    d1 = {'zero':'0', 'one':'1', 'two':'2', 'three':'3', 'four':'4', 
         'five':'5', 'six':'6', 'seven':'7', 'eight':'8', 'nine':'9'}
    str1 = ""
    visited = [False] * len(s)

    for i in range(len(s)):
        if not visited[i]:
            if ord('0') <= ord(s[i]) <= ord('9'):
                str1 += s[i]
                visited[i] = True
            else:
                str2 = ""
                for j in range(i, len(s)):
                    if ord('a') <= ord(s[j]) <= ord('z'):
                        str2 += s[j]
                        visited[j] = True
                    if str2 in d1:
                        str1 += d1[str2]
                        break
    answer = int(str1)
    return answer

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;

class Solution {
    public HashMap<String, String> hm;
    
    public int solution(String s) {
        hm = new HashMap<>();
        
        hm.put("zero", "0");
        hm.put("one", "1");
        hm.put("two", "2");
        hm.put("three", "3");
        hm.put("four", "4");
        hm.put("five", "5");
        hm.put("six", "6");
        hm.put("seven", "7");
        hm.put("eight", "8");
        hm.put("nine", "9");
        
        for (String tmp : hm.keySet()) {
            s = s.replace(tmp, hm.get(tmp));
        }
        
        return Integer.parseInt(s);
    }
}